Source/Scene/PointCloudEyeDomeLighting.js

```diff
@@ -0,0 +1,400 @@
+            });
+
+            vs.sources.push(
+                'varying vec3 v_positionECPS; \n' +
+                'void main() \n' +
+                '{ \n' +
+                '    czm_point_cloud_post_process_main(); \n' +
+                '    v_positionECPS = (czm_inverseProjection * gl_Position).xyz; \n' +
+                '}');
+            fs.sources.unshift('#extension GL_EXT_draw_buffers : enable \n');
+            fs.sources.push(
+                'varying vec3 v_positionECPS; \n' +
+                'void main() \n' +
+                '{ \n' +
+                '    czm_point_cloud_post_process_main(); \n' +
+                // Write log base 2 depth to alpha for EDL
+                '    gl_FragData[1] = vec4(v_positionECPS, log2(-v_positionECPS.z)); \n' +
+                '}');
+
+            shader = context.shaderCache.createDerivedShaderProgram(shaderProgram, 'EC', {
+                vertexShaderSource : vs,
+                fragmentShaderSource : fs,
+                attributeLocations : attributeLocations
+            });
+        }
+
+        return shader;
+    }
+
+    PointCloudEyeDomeLighting.prototype.update = function(frameState, commandStart, tileset) {
+        var passes = frameState.passes;
+        var isPick = (passes.pick && !passes.render);
+        if (!processingSupported(frameState.context) || isPick) {
+            return;
+        }
+
+        this._strength = tileset.pointShading.eyeDomeLightingStrength;
+        this._radius = tileset.pointShading.eyeDomeLightingRadius;
+
+        var dirty = createResources(this, frameState.context, false);
+
+        // Hijack existing point commands to render into an offscreen FBO.
+        var i;
+        var commandList = frameState.commandList;
+        var commandEnd = commandList.length;
+
+        for (i = commandStart; i < commandEnd; ++i) {
+            var command = commandList[i];
+            if (command.primitiveType !== PrimitiveType.POINTS) {
+                continue;
+            }
+            var derivedCommand = command.derivedCommands.pointCloudProcessor;
+            if (!defined(derivedCommand) || command.dirty || dirty ||
+                (derivedCommand.framebuffer !== this._framebuffers.prior)) { // Prevent crash when tiles out-of-view come in-view during context size change
+                derivedCommand = DrawCommand.shallowClone(command);
+                command.derivedCommands.pointCloudProcessor = derivedCommand;
+
+                derivedCommand.framebuffer = this._framebuffers.prior;
+                derivedCommand.shaderProgram = getECShaderProgram(frameState.context, command.shaderProgram);
+                derivedCommand.castShadows = false;
+                derivedCommand.receiveShadows = false;
+
+                var derivedCommandRenderState = clone(derivedCommand.renderState);
+                derivedCommand.renderState = RenderState.fromCache(
+                    derivedCommandRenderState
+                );
```

Comment:
The render state isn't being modified so I don't think these couple of lines are needed.